### PR TITLE
chore: upgrade ui-kit to v12

### PR DESCRIPTION
**Pull request by adnasa**
#### Update

> ℹ️ This now targets https://github.com/commercetools/merchant-center-application-kit/pull/2041

#### Summary

- upgrades ui-kit to canary version of v12 (canary)
- adjusts components to remove use of deprecated props

#### Description

I opened this PR to check that the build passes and inspect the Percy snapshots.
If things are fine, I will:

1. release ui-kit v12
2. update this PR (versions and title)
3. changeset

### CODE REVIEW

File: @types/commercetools-uikit__design-system/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+declare module '@commercetools-uikit/design-system' {
+  export const customProperties: {
+    [key: string]: string;
+  };
+}
```
Comment (adnasa) [inline @types/commercetools-uikit__design-system/index.d.ts:5]: doing a `yarn typecheck` yielded errors, so I added this.
My intuition is that it doesn't make sense to do this, when the package itself was written in TS.
but I had a brainfart and couldn't figure out where to start addressing this properly.

File: packages/application-components/src/components/modal-pages/internals/modal-page-top-bar.tsx [hunk 2]
```diff
@@ -96,8 +96,8 @@ const ModalPageTopBar = (props: Props) => {
         />
         {props.currentPathLabel && (
           <React.Fragment>
-            <Text.Detail isInline>/</Text.Detail>
-            <Text.Detail title={props.currentPathLabel} isInline truncate>
```
Comment (adnasa) [inline packages/application-components/src/components/modal-pages/internals/modal-page-top-bar.tsx:100]: ooooops! 
this has to be `as="span"`. 
what do you think?

File: packages/application-components/src/components/modal-pages/internals/modal-page-top-bar.tsx [hunk 3]
```diff
@@ -96,8 +96,8 @@ const ModalPageTopBar = (props: Props) => {
         />
         {props.currentPathLabel && (
           <React.Fragment>
-            <Text.Detail isInline>/</Text.Detail>
```
Comment (adnasa) [inline packages/application-components/src/components/modal-pages/internals/modal-page-top-bar.tsx:99]: wait. why is this here?

**Comment by adnasa** [comment]
this now points of `v19`

File: packages/application-components/src/components/modal-pages/internals/modal-page-top-bar.tsx (hunk 2 above)
Comment (adnasa) [reply packages/application-components/src/components/modal-pages/internals/modal-page-top-bar.tsx:100]: fixed via https://github.com/commercetools/ui-kit/pull/1854

File: .changeset/small-grapes-cross.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (adnasa) [inline .changeset/small-grapes-cross.md:1]: some of these packages were still on v17, @emmenko 
Shall we generate a "final" changeset, setting all packages to be v19?
what do you think?

**Comment by adnasa** [comment]
oohh yeah, build passed 😎

File: .changeset/small-grapes-cross.md (hunk 4 above)
Comment (emmenko) [reply .changeset/small-grapes-cross.md:1]: Sure, we can do that.

File: @types/commercetools-uikit__design-system/index.d.ts (hunk 1 above)
Comment (emmenko) [reply @types/commercetools-uikit__design-system/index.d.ts:5]: I can take a look, but this file should be removed.

**Review by emmenko — APPROVED** [review]
